cg/store/api/find_business_data.py

```diff
@@ -30,39 +30,21 @@
 from cg.store.filters.status_flow_cell_filters import apply_flow_cell_filter, FlowCellFilter
 from cg.store.filters.status_case_sample_filters import apply_case_sample_filter, CaseSampleFilter
 from cg.store.filters.status_sample_filters import apply_sample_filter, SampleFilter
-
+from cg.store.filters.status_analysis_filters import apply_analysis_filter, AnalysisFilter
 
 LOG = logging.getLogger(__name__)
 
 
 class FindBusinessDataHandler(BaseHandler):
     """Contains methods to find business data model instances"""
 
-    def analyses(self, *, family: Family = None, before: dt.datetime = None) -> Query:
-        """Fetch multiple analyses."""
-        records = self.Analysis.query
-        if family:
-            query_family = family
-            records = records.filter(Analysis.family == query_family)
-        if before:
-            subq = (
-                self.Analysis.query.join(Analysis.family)
-                .filter(Analysis.started_at < before)
-                .group_by(Family.id)
-                .with_entities(
-                    Analysis.family_id,
-                    func.max(Analysis.started_at).label("started_at"),
-                )
-                .subquery()
-            )
-            records = records.join(
-                subq,
-                and_(
-                    self.Analysis.family_id == subq.c.family_id,
-                    self.Analysis.started_at == subq.c.started_at,
-                ),
-            ).filter(Analysis.started_at < before)
-        return records
+    def get_analyses_by_case(self, case: Family) -> List[Analysis]:
```

Comment:
Should we really pass objects to the filter functions or just the properties? I think it is better to just pass the property value - in this case the case_entry_id

Reply:
Fixed